snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -412,6 +413,54 @@ def write_snap_directory(self) -> None:
 
         self._record_manifest_and_source_snapcraft_yaml()
 
+    def warn_ld_library_paths(self) -> None:
+        root_environment = self._config_data.get("environment", dict())
+        root_ld_library_path = root_environment.get("LD_LIBRARY_PATH")
+        # Dictionary of app names with LD_LIBRARY_PATH in their environment.
+        app_environment: Dict[str, str] = dict()
+
+        for app_name, app_props in self._config_data.get("apps", dict()).items():
+            with contextlib.suppress(KeyError):
+                app_environment[app_name] = app_props["environment"]["LD_LIBRARY_PATH"]
+
+        if root_ld_library_path is None and not app_environment:
+            return
+
+        ld_library_path_empty: Set[str] = set()
+        if root_ld_library_path is None and app_environment:
+            ld_library_path_empty = {
+                name
+                for name, ld_env in app_environment.items()
+                if "$LD_LIBRARY_PATH" in ld_env
```

Comment:
I don't think this would catch `${LD_LIBRARY_PATH}` which I believe is valid?

Reply:
done